### PR TITLE
[FIX]: copy pasted SVG or PNG cannot be undone

### PR DESCRIPTION
**Pull request by vedang685**
# Fix 
Copy pasted SVG or PNG cannot be undone 
Issue #8321 

# Video of Fixed bug  

https://github.com/user-attachments/assets/13c289b4-2a85-4462-b4a7-d432af582a89

### CODE REVIEW

File: packages/excalidraw/components/App.tsx [hunk 1]
```diff
@@ -2988,6 +2991,8 @@ class App extends React.Component<AppProps, AppState> {
           ),
         });
 
+        this.actionManager.executeAction(actionFinalize);
```
Comment (mtolmacs) [inline packages/excalidraw/components/App.tsx:2994]: This removes the selection from the freshly pasted image, which leads to test failures. You can run the tests locally to see by `yarn test:app`.

The expected case is that after pasting the pasted image is selected.

**Review by mtolmacs — CHANGES_REQUESTED** [review]
Thanks for the PR. Please see below an issue pinpointed by the test suite.

File: packages/excalidraw/components/App.tsx (hunk 1 above)
Comment (vedang685) [reply packages/excalidraw/components/App.tsx:2994]: Sure, that makes sense. I'll implement the changes

**Comment by vedang685** [comment]
@mtolmacs I guess the issue is resolved, Attached the video as well. Can you check?

File: packages/excalidraw/components/App.tsx [hunk 2]
```diff
@@ -1950,6 +1950,10 @@ class App extends React.Component<AppProps, AppState> {
         frameElement,
         data: { status: "done", html: parsedHtml },
       });
+
```
Comment (mtolmacs) [inline packages/excalidraw/components/App.tsx:1953]: What's the point of these lines?

**Review by mtolmacs — CHANGES_REQUESTED** [review]

File: packages/excalidraw/components/App.tsx (hunk 2 above)
Comment (vedang685) [reply packages/excalidraw/components/App.tsx:1953]: @mtolmacs Are you referring for the extra line I have inserted between line 1952 and 1954?

Comment (mtolmacs) [reply packages/excalidraw/components/App.tsx:1953]: No, I was referring to this addition with setState. Why did you add that?

Comment (vedang685) [reply packages/excalidraw/components/App.tsx:1953]: @mtolmacs it is being used to update the selected Element ID after new frame is created, so the newly pasted frame gets selected

Comment (mtolmacs) [reply packages/excalidraw/components/App.tsx:1953]: This might be a misunderstanding, the `frameElement` here is something unrelated, it has to do with the frame tool.

Comment (vedang685) [reply packages/excalidraw/components/App.tsx:1953]: @mtolmacs oh yeah, you're correct. Thankyou for pointing our
Pushing the changes

**Comment by mtolmacs** [comment]
@Mrazator I think this PR is ready for your review and solves the issue properly.

**Review by Mrazator — CHANGES_REQUESTED** [review]
Nice! Few things:

- it does not work for drag & drop
- add a simple test cases as part of the `history.test.tsx` to have the main cases covered (at least for all the copy paste cases, but would be nice also for drag & drop)
- make sure all the references of `insertImageElement` that should be recorded are also recorded (there is a fishy usage as part of `mixedContent`)
- go through related copy / paste to find similar cases we missed:
  - i.e. pasting embeddable like ty link creates iframe, but similarly can't be undone
- go through related drag & drop edge cases
  - i.e. pasting mermaid, spreadsheet data and svg works

**Comment by vedang685** [comment]
> Nice! Few things:
> 
> * it does not work for drag & drop
> * add a simple test cases as part of the `history.test.tsx` to have the main cases covered (at least for all the copy paste cases, but would be nice also for drag & drop)
> * make sure all the references of `insertImageElement` that should be recorded are also recorded (there is a fishy usage as part of `mixedContent`)
> * go through related copy / paste to find similar cases we missed:
>   
>   * i.e. pasting embeddable like yt link creates iframe, but similarly can't be undone
> * go through related drag & drop edge cases
>   
>   * i.e. pasting mermaid, spreadsheet data, and svg can be undone

@Mrazator Sure! Are these issues also related to recording the paste in history store?

**Comment by Mrazator** [comment]
@vedang685 thanks for the effort, though in the end we've realized it's not a good first issue at all, so we've solved it as part of #9605
